### ISSUE TITLE
Doesn't work on Hyper 3 unless you disable WebGL rendering

### ISSUE THREAD

**Issue by cdhagmann** [tolbertam/hyper-font-ligatures#8]

**Comment by tolbertam** [comment]
Thanks for the report.  I tried hyper 3 out and it looks like `onDecorated` is no longer being exposed (EDIT: You can probably ignore this statement, that doesn't seem unusual).  I'm traveling today, but I'll look at it tomorrow.

**Comment by tolbertam** [comment]
Possibly related? xtermjs/xterm.js#2028

I'll try a local build of hyper with xterm with that fix and see how it goes! :crossed_fingers:

**Comment by tolbertam** [comment]
This may be a bit trickier than anticipated.  It looks like hyper is using a fork of xterm.js (https://github.com/zeit/xterm.js),  I tried merging recent xterm.js changes into it and doing a custom build, but it's not much of a trivial exercise.

The aforementioned issue I mentioned probably isn't related, since the hyper fork is based on xterm 3.12.0 with the WebGL renderer work that is not yet merged into xterm proper, and the issue was opened about a regression in 3.13 beta.

This plugin itself doesn't do much, it simply uses `xterm-addon-ligatures`, so we'll mostly be at the mercy of xterm.  I suspect it's possible that the WebGL renderer isn't currently taking into account ligatures, and that project has done an excellent job supporting ligatures in the first place, so I bet at some point this plugin will start working again :crossed_fingers: .  I'll add a note to the README that this plugin is currently broken on Hyper 3 and also keep an eye on xterm development.

**Comment by r4dixx** [comment]
FYI this workaround seems to do the trick on Windows. Just tested it on macOS, works fine too. https://github.com/zeit/hyper/issues/3607#issuecomment-489787741

**Comment by tolbertam** [comment]
Thanks @r4dixx, didn't see that issue as it was closed before I opened up this one.  Good to have that as a cross reference, will update the docs to reflect this.

**Comment by lackdaz** [comment]
@r4dixx Just to be certain, adding `webGLRenderer: false` under your config: with this plugin installed brings back ligatures?

I'm running MacOSX and this doesn't seem to be the fix

**Comment by r4dixx** [comment]
@lackdaz Yup, works for me at least. I cannot tell for sure if it works for others. I'm running the very latest stable versions of both macOS and Hyper. Works for me with this plugin installed. Have you performed a full reload?

**Comment by lackdaz** [comment]
I’ve been putting off upgrading my macOSX (10.12.6) because of some vm related issues with the latest versions of macOSX. Perhaps this might be a push. Yes I’ve done a full reload, are there any css parameters that you’ve put in too? I’m here because the x-row, x-screen css deprecation

**Comment by r4dixx** [comment]
@lackdaz I didn't add/change anything in `.hyper.js` except:

```
module.exports = {
  config: {
    webGLRenderer: false
  },
  plugins: [
    "hyper-font-ligatures"
  ],
};
```

No custom `css`, `termCSS`, or anything, sorry.

Hyper 3.0.2 (stable)
hyper-font-ligatures (0.0.7)

And macOS Mojave 10.14.5 (18F203) but I wouldn't know if it is related to your OS version or not.

**Comment by lackdaz** [comment]
That's useful information. I'll try it out and keep you posted! Thanks!

I'm sorely missing my ligatures

**Comment by tolbertam** [comment]
I tested this on Linux and it looks to work there as well!  I updated the README with the guidance provided here and published a 0.0.8 release (so documentation on npm includes this).

I'll leave this open for the time being, as I imagine others will encounter this and seeing an already open issue may help point them in the right direction.

**Comment by lackdaz** [comment]
It was my mistake, I upgraded to Mojave but I had the `webGLRenderer: false` beneath `plugins` and hence not within `config`

I can confirm that this is fixed now and I would willingly trade ligatures for kernel cycles 👯‍♂ 

Doh!

**Comment by Tylerwbrown** [comment]
For me disabling the webGLRenderer does bring back ligatures, but has a significant impact on performance. I'm not able to do things that were possible before. 

I noticed this after having 2 vim tabs open and trying to open a third to run something, it started being incredibly slow, even just opening a new hyper window and spamming new terminal tabs starts being incredibly slow after 4 tabs open, but not with webGLRenderer enabled.

**Comment by tolbertam** [comment]
> For me disabling the webGLRenderer does bring back ligatures, but has a significant impact on performance. I'm not able to do things that were possible before.

I recently noticed precisely the same, particularly when using vim as you mentioned, and also doing some other interactive tasks.  I think the plugin is a bit at the mercy of hyper & xterm here.

**Comment by zoedsoupe** [comment]
I'd disabled webGLRenderer but ligatures doesn't work :/

**Comment by jacob-israel-turner** [comment]
On a fresh install, it doesn't appear that disabling WebGL is required anymore!

EDIT: Scratch that - turns out WebGL rendering is turned off by default on new Hyper installations.  Turning it on does indeed break this plugin.

**Comment by jacob-israel-turner** [comment]
Hey guys!  I just introduced a PR that should resolve this issue, to an extent:
https://github.com/tolbertam/hyper-font-ligatures/pull/11

There are still known issues with WebGL and Ligatures, but in my experience it's still very usable.

**Comment by BitInByte** [comment]
Hey guys, there are any plans to fix this in hyper 3 with WebGL enabled? I would like to have it :)

**Comment by ramarivera** [comment]
Plus to this request :)
